### PR TITLE
Fix `cupy.copyto` to take NumPy array scalars

### PR DESCRIPTION
**Pull request by takagi**
Rel #6577.

This PR fixes `cupy.copyto` to be able to take NumPy array scalars as its `src` data. I want to change this in advance to fixing #6577 in order to implement `cupy.full` using `cupy.copyto` as `numpy.full` does.

### CODE REVIEW

File: cupy/_manipulation/basic.py [hunk 1]
```diff
@@ -35,6 +36,14 @@ def copyto(dst, src, casting='same_kind', where=None):
     if src_is_python_scalar:
         src_dtype = numpy.dtype(type(src))
         can_cast = numpy.can_cast(src, dst.dtype, casting)
+    elif isinstance(src, numpy.ndarray):
+        if src.shape != ():
```
Comment (asi1024) [inline cupy/_manipulation/basic.py:40]: ndarray.item works when src.size == 1?

Comment (takagi) [reply cupy/_manipulation/basic.py:40]: Yes, passing no arguments to ndrray.item is for that purpose. (Or you say another thing?)

https://numpy.org/doc/stable/reference/generated/numpy.ndarray.item.html
> none: in this case, the method only works for arrays with one element (a.size == 1), which element is copied into a standard Python scalar object and returned.

Comment (asi1024) [reply cupy/_manipulation/basic.py:40]: How about changing this condition to `src.size == 1`? We can support also `(1,)` shape arrays.

Comment (takagi) [reply cupy/_manipulation/basic.py:40]: Sure!

**Review by asi1024 — APPROVED** [review]

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**